arxiv/document/metadata.py

```diff
@@ -0,0 +1,313 @@
+"""Representations of arXiv document metadata."""
+from collections import abc
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Iterator, List, Optional, Set, Literal, Sequence
+
+from ..taxonomy import definitions
+from ..taxonomy.category import Category, Group, Archive
+from ..identifier import Identifier
+from ..license import License
+from .version import VersionEntry
+
+from ..base.urls import canonical_url
+
+
+@dataclass(frozen=True)
```

Comment:
I've found frozen has been unpleasant here because prevents the creation of subclasses. 

No change needed now.